solr/core/src/java/org/apache/solr/search/FunctionQParser.java

```diff
@@ -231,7 +232,59 @@ public String parseArg() throws SyntaxError {
     return val;
   }
 
-  
+  /**
+   * Parse argument in the same way as parseArg() but also parse the argument name
+   * if written in following syntax: argumentName=argument.
+   *
+   * @return Immutable entry with name as the key and argument as value. In case where there's no name, key is null.
+   * @throws SyntaxError in case when argument is not ended by ) or ,
+   */
+  public SimpleImmutableEntry<String, String> parseNamedArg() throws SyntaxError {
```

Comment:
1. Usually it's nicer to declare interface ie Entry<>.
1. I've thought that Yonik whiteness about something already existed? Doesn't it?

Reply:
1. Agree. 
2. [Here](https://issues.apache.org/jira/browse/SOLR-12325?focusedCommentId=16649083&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16649083) he said "I wish we had..." - i.e. it did't exist at least at that time. And I can't find it now...